orangecontrib/bioinformatics/widgets/OWMarkerGenes.py

```diff
@@ -233,222 +890,245 @@ def data(self, value: Table):
+        )
+        self.selected_markers_view.model().sourceModel().data_added.connect(self._selected_markers_changed)
+        self.selected_markers_view.model().sourceModel().data_removed.connect(self._selected_markers_changed)
 
-    def update_data_info(self):
-        model = self.view.model()
-        self.info.set_input_summary(f"Shown : {str(len(model.source))}/{str(model.data_length())}")
-        self.info.set_output_summary(f"Selected: {str(len(self.selected_genes))}")
+        # update output and messages
+        self._selected_markers_changed()
 
-    def _setup(self):
-        self.closeContext()
-        data = self.data
+    def _filter_data_group(self, data: Table) -> Tuple[Table, Tuple]:
+        """
+        Function filter the table based on the selected group (Mouse, Human) and divide them in two groups based on
+        selected_data variable.
+
+        Parameters
+        ----------
+        data
+            Table to be filtered
+
+        Returns
+        -------
+        data_not_selected
+            Data that will initially be in available markers view.
+        data_selected
+            Data that will initially be in selected markers view.
+        """
         group = data.domain.metas[0]
         gvec = data.get_column_view(group)[0]
 
         if group.is_string:
-            mask = gvec == self.group_cb.itemData(self.group_index, Qt.DisplayRole)
+            mask = gvec == self.selected_organism
         else:
-            mask = gvec == self.group_index
-
+            mask = gvec == self.organism_index
         data = data[mask]
-        rest = data[:, data.domain.metas[1:]]
-        model = SearchableTableModel(rest, parent=self)
-        ref_col = rest.domain.metas.index(rest.domain[HeaderLabels[HeaderIndex.REFERENCE]])
-        self.view.setItemDelegateForColumn(ref_col, gui.LinkStyledItemDelegate(self.view))
-
-        self.view.setModel(model)
-        self.view.selectionModel().selectionChanged.connect(self._on_selection_changed)
 
-        self.openContext(self.selected_group)
-        self.call_filter_timer(self.filter_text)
-        self.view.hideColumn(HeaderIndex.URL)
-
-    def _on_selection_changed(self, *args):
-        self.update_model()
-        self.commit()
-        self.update_data_info()
+        # divide data based on selected_genes variable (context)
+        unique_gene_names = np.core.defchararray.add(
+            data.get_column_view("Entrez ID")[0].astype(str), data.get_column_view("Cell Type")[0].astype(str)
+        )
+        mask = np.isin(unique_gene_names, self.selected_genes)
+        data_not_selected = data[~mask]
+        data_selected = data[mask]
+        return data_not_selected, data_selected
+
+    def commit(self) -> None:
+        rows = self.selected_markers_view.model().sourceModel().rootItem.get_data_rows()
+        if len(rows) > 0:
+            metas = [r.metas for r in rows]
+            data = Table(self.data.domain, np.empty((len(metas), 0)), metas=metas)
```

Comment:
Calls to Table constructor are deprecated: https://github.com/biolab/orange3/commit/76846e2817e36ea24966344e1de27a2be7310a79